src/eventDispatchers/mouseEventHandlers/mouseDown.js

```diff
@@ -88,7 +88,22 @@ export default function(evt) {
 
   if (annotationToolsWithMoveableHandles.length > 0) {
     const firstToolWithMoveableHandles = annotationToolsWithMoveableHandles[0];
-    const toolState = getToolState(element, firstToolWithMoveableHandles.name);
+    let toolState = getToolState(element, firstToolWithMoveableHandles.name);
+
+    // create a deep copy of toolState to avoid bad renders of contours after right clicks
+    toolState = JSON.parse(JSON.stringify(toolState));
+
+    const activeRoiId = firstToolWithMoveableHandles.roi.id;
+    let temporaryValue;
+
+    // Moves the elements with a roi id equal to the activeRoiId to the beginning of the toolState array
+    for (let i = 0; i < toolState.data.length; i++) {
+      if (toolState.data[i].roi.id === activeRoiId) {
```

Comment:
It seems that changing this order causes existing contours to not be able to be deleted in `nota`:

```
Uncaught TypeError: Cannot destructure property 'wadoUrl' of 'instancesById[affectedInstanceId]' as it is undefined.
    at :3000/app/static/js/main.chunk.js:41757
    at :3000/app/static/js/0.chunk.js:553650
    at :3000/app/static/js/0.chunk.js:284317
    at deleteContourToolData (:3000/app/static/js/main.chunk.js:52596)
    at onContourDeleteClicked (:3000/app/static/js/main.chunk.js:52157)
    at HTMLUnknownElement.callCallback (:3000/app/static/js/0.chunk.js:513136)
    at Object.invokeGuardedCallbackDev (:3000/app/static/js/0.chunk.js:513185)
    at invokeGuardedCallback (:3000/app/static/js/0.chunk.js:513245)
    at invokeGuardedCallbackAndCatchFirstError (:3000/app/static/js/0.chunk.js:513260)
    at executeDispatch (:3000/app/static/js/0.chunk.js:517495)
    at processDispatchQueueItemsInOrder (:3000/app/static/js/0.chunk.js:517527)
    at processDispatchQueue (:3000/app/static/js/0.chunk.js:517540)
    at dispatchEventsForPlugins (:3000/app/static/js/0.chunk.js:517551)
    at :3000/app/static/js/0.chunk.js:517762
    at batchedEventUpdates$1 (:3000/app/static/js/0.chunk.js:531447)
    at batchedEventUpdates (:3000/app/static/js/0.chunk.js:512934)
    at dispatchEventForPluginEventSystem (:3000/app/static/js/0.chunk.js:517761)
    at attemptToDispatchEvent (:3000/app/static/js/0.chunk.js:515244)
    at dispatchEvent (:3000/app/static/js/0.chunk.js:515162)
    at unstable_runWithPriority (:3000/app/static/js/0.chunk.js:555577)
    at runWithPriority$1 (:3000/app/static/js/0.chunk.js:520542)
    at discreteUpdates$1 (:3000/app/static/js/0.chunk.js:531464)
    at discreteUpdates (:3000/app/static/js/0.chunk.js:512946)
    at dispatchDiscreteEvent (:3000/app/static/js/0.chunk.js:515128)
```

Reply:
Completely forgot to test "Delete."

Reply:
Making the deep copy caused the error, so I removed that. Maybe it might be better to "set" the toolState instead of having the state changed by reference... should I do that or leave it as is?